### PR TITLE
Add QR code to long lived access tokens dialog

### PR DESCRIPTION
**Pull request by ZephireNZ**
## Proposed change

This change adds a QR code when generating an long lived access token. This can then be used by, for example, a mobile app to load the token without having to rely on the user to copy/paste this manually.

image
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR is related to issue or discussion: https://community.home-assistant.io/t/show-long-lived-access-tokens-as-qr-code/288016

Code largely copied from `dialog-tag-detail.ts`, it may be worthwhile to refactor code to use a common helper function.

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        <div>
+          <p class="no-bottom-padding">
+            ${this.hass.localize(
+              "ui.panel.profile.long_lived_access_tokens.prompt_copy_token"
+            )}
+          </p>
+          <paper-input
+            dialogInitialFocus
+            .value=${this._params.token}
+            .type=${"text"}
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:61]:
```suggestion
            type="text"
```

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+          ></paper-input>
+          <div id="qr">
+            ${this._qrCode
+              ? this._qrCode
+              : html`
+                  <mwc-button @click=${this._generateQR}
+                    >Generate QR code
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:68]:
```suggestion
                  <mwc-button @click=${this._generateQR}>
                    Generate QR code
```

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+          padding-bottom: 24px;
+          color: var(--primary-text-color);
+        }
+        .no-bottom-padding {
+          padding-bottom: 0;
+        }
+        ha-dialog {
+          /* Place above other dialogs */
+          --dialog-z-index: 104;
+        }
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:124]: We don't open this together with other dialogs right? So this is not needed.

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  }
+
+  static get styles(): CSSResult[] {
+    return [
+      haStyleDialog,
+      css`
+        :host([inert]) {
+          pointer-events: initial !important;
+          cursor: initial !important;
+        }
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:108]: What is this for?

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        hideActions
+        .heading=${createCloseHeading(this.hass, this._params.name)}
+        @closed=${this.closeDialog}
+      >
+        <div>
+          <p class="no-bottom-padding">
+            ${this.hass.localize(
+              "ui.panel.profile.long_lived_access_tokens.prompt_copy_token"
+            )}
+          </p>
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:57]: Should this be a label on the input?

File: src/panels/profile/ha-long-lived-access-token-dialog.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        a {
+          color: var(--primary-color);
+        }
+        p {
+          margin: 0;
+          padding-top: 6px;
+          padding-bottom: 24px;
+          color: var(--primary-text-color);
+        }
```
Comment (bramkragten) [inline src/panels/profile/ha-long-lived-access-token-dialog.ts:117]: I don't think this is needed, as it should be provided by `haStyleDialog`

File: src/panels/profile/ha-long-lived-access-token-dialog.ts (hunk 4 above)
Comment (ZephireNZ) [reply src/panels/profile/ha-long-lived-access-token-dialog.ts]: I had copied the CSS from ha-dialog, removing it doesn't appear to effect the dialog so have removed it.

File: src/panels/profile/ha-long-lived-access-token-dialog.ts (hunk 5 above)
Comment (ZephireNZ) [reply src/panels/profile/ha-long-lived-access-token-dialog.ts]: Changed 😊

File: src/panels/profile/ha-long-lived-access-token-dialog.ts (hunk 6 above)
Comment (ZephireNZ) [reply src/panels/profile/ha-long-lived-access-token-dialog.ts]: This was copied from ha-dialog, you're correct (especially now that I have removed the `<p>` element), removed.

File: src/panels/profile/ha-long-lived-access-token-dialog.ts (hunk 3 above)
Comment (ZephireNZ) [reply src/panels/profile/ha-long-lived-access-token-dialog.ts]: Yup, was copied over from ha-dialog but agree, not needed. Removed.

**Comment by APayerl** [comment]
What has happened to this PR? It has been a while now without any activity as far as I can see?

**Review by bramkragten — APPROVED** [review]

**Comment by bramkragten** [comment]
> What has happened to this PR? It has been a while now without any activity as far as I can see?

Oops 😅

**Pull request merged**